src/pastelid/pastel_key.cpp

```diff
@@ -0,0 +1,314 @@
+                const auto sFilePath = GetKeyFilePath(sPastelID);
+                // read public items from the secure container file
+                if (!cont.read_public_from_file(sFilePath))
+                    throw std::runtime_error("Cannot verify signature with LegRoast algorithm. LegRoast public key is not generated");
+                std::string sLegRoastPubKey;
+                v_uint8 vLRPubKey;
+                // retrieve encoded LegRoast public key
+                if (!cont.get_public_data(PUBLIC_ITEM_TYPE::pubkey_legroast, sLegRoastPubKey))
+                    throw std::runtime_error("Cannot verify signature with LegRoast algorithm. LegRoast public key associated with the PastelID was not found");
+                // decode base58-encoded LegRoast public key
+                std::string error;
+                if (DecodeLegRoastPubKey(sLegRoastPubKey, vLRPubKey))
+                {
+                    bool bValid = false;
+                    // verify signature
+                    CLegRoast<algorithm::Legendre_Middle> LegRoast;
+                    if (LegRoast.set_public_key(error, vLRPubKey.data(), vLRPubKey.size()))
+                    {
+	                    if (fBase64)
+        	                bValid = LegRoast.set_signature(error, ed_crypto::Base64_Decode(sSignature));
+                	    else
+	                        bValid = LegRoast.set_signature(error, reinterpret_cast<const unsigned char*>(sSignature.data()), sSignature.size());
+                    }
+                    if (!bValid)
+                    	throw std::runtime_error(strprintf("Cannot verify signature with LegRoast algorithm. %s", error));
+                    bRet = LegRoast.verify(error, reinterpret_cast<const unsigned char*>(sText.data()), sText.size());
+                }
+            } break;
+
+            default:
+                break;
+        } // switch
+    } catch (const ed_crypto::crypto_exception& ex) {
+        throw std::runtime_error(ex.what());
+    }
+    return bRet;
+}
+
+/**
+* Get PastelIDs stored locally in pastelkeys (pastelkeysdir option).
+* 
+* \param bPastelIdOnly - return PastelIDs only, otherwise returns PastelIDs along with associated keys
+*                        read from the secure container
+* \return map of PastelID -> associated keys (LegRoast signing public key)
+*/
+pastelid_store_t CPastelID::GetStoredPastelIDs(const bool bPastelIdOnly)
+{
+    using namespace secure_container;
+    fs::path pathPastelKeys(GetArg("-pastelkeysdir", "pastelkeys"));
+    pathPastelKeys = GetDataDir() / pathPastelKeys;
+
+    pastelid_store_t resultMap;
+    std::string sPastelID, sLegRoastKey;
+    v_uint8 vData;
+    for (const auto& p : fs::directory_iterator(pathPastelKeys))
+    {
+        sPastelID = p.path().filename().string();
+        // check if this file name is in fact encoded Pastel ID
+        if (!DecodePastelID(sPastelID, vData))
+            continue;
+        sLegRoastKey.clear();
```

Comment:
and this can be removed then
sLegRoastKey.clear();

Reply:
If I reuse any variable in a loop - I like to clean it up before each usage. You never know who will be changing this function and how this variable will be used in this loop later on. 
So cleanup seems to me like a good idea. 
Does not matter that in this case this string is moved to the pastelid storage. 
IMHO this contributes to safe and reliable code.

Reply:
ok. good